azkaban-common/src/test/java/azkaban/executor/ExecutableRampDependencyMapTest.java

```diff
@@ -0,0 +1,132 @@
+    executableMap.add(DEPENDENCY_NAME6, DEPENDENCY_VALUE21, null);
+    Assert.assertEquals(5, executableMap.size());
+    Assert.assertTrue(executableMap.isValidJobType(DEPENDENCY_NAME1, JOB_TYPE_SPARK));
+    Assert.assertFalse(executableMap.isValidJobType(DEPENDENCY_NAME1, JOB_TYPE_PIG));
+    Assert.assertFalse(executableMap.isValidJobType(DEPENDENCY_NAME2, JOB_TYPE_SPARK));
+    Assert.assertTrue(executableMap.isValidJobType(DEPENDENCY_NAME2, JOB_TYPE_PIG));
+    Assert.assertTrue(executableMap.isValidJobType(DEPENDENCY_NAME3, JOB_TYPE_SPARK));
+    Assert.assertTrue(executableMap.isValidJobType(DEPENDENCY_NAME3, JOB_TYPE_PIG));
+    Assert.assertFalse(executableMap.isValidJobType(DEPENDENCY_NAME3, JOB_TYPE_MR));
+    Assert.assertTrue(executableMap.isValidJobType(DEPENDENCY_NAME5, JOB_TYPE_SPARK));
+    Assert.assertTrue(executableMap.isValidJobType(DEPENDENCY_NAME5, JOB_TYPE_PIG));
+    Assert.assertTrue(executableMap.isValidJobType(DEPENDENCY_NAME5, JOB_TYPE_MR));
+    Assert.assertTrue(executableMap.isValidJobType(DEPENDENCY_NAME6, JOB_TYPE_SPARK));
+    Assert.assertTrue(executableMap.isValidJobType(DEPENDENCY_NAME6, JOB_TYPE_PIG));
+    Assert.assertTrue(executableMap.isValidJobType(DEPENDENCY_NAME6, JOB_TYPE_MR));
+    Assert.assertFalse(executableMap.isValidJobType(DEPENDENCY_NAME0, JOB_TYPE_SPARK));
+  }
+
+  @Test
+  public void testUploadDependency() {
+    executableMap.add(DEPENDENCY_NAME1, DEPENDENCY_VALUE11, JOB_TYPE_SPARK);
+    Assert.assertEquals(1, executableMap.size());
+    Assert.assertEquals(DEPENDENCY_VALUE11, executableMap.getDefaultValue(DEPENDENCY_NAME1));
+    Assert.assertTrue(executableMap.isValidJobType(DEPENDENCY_NAME1, JOB_TYPE_SPARK));
+    Assert.assertFalse(executableMap.isValidJobType(DEPENDENCY_NAME1, JOB_TYPE_MR));
+    Assert.assertFalse(executableMap.isValidJobType(DEPENDENCY_NAME1, JOB_TYPE_PIG));
+
+    ExecutableRampDependencyMap novaExecutableMap = ExecutableRampDependencyMap.createInstance();
+    novaExecutableMap.add(DEPENDENCY_NAME1, DEPENDENCY_VALUE12, JOB_TYPE_SPARK_PIG);
+    executableMap.refresh(novaExecutableMap);
+    Assert.assertEquals(1, executableMap.size());
+    Assert.assertNotEquals(DEPENDENCY_VALUE11, executableMap.getDefaultValue(DEPENDENCY_NAME1));
+    Assert.assertEquals(DEPENDENCY_VALUE12, executableMap.getDefaultValue(DEPENDENCY_NAME1));
+    Assert.assertTrue(executableMap.isValidJobType(DEPENDENCY_NAME1, JOB_TYPE_SPARK));
+    Assert.assertFalse(executableMap.isValidJobType(DEPENDENCY_NAME1, JOB_TYPE_MR));
+    Assert.assertTrue(executableMap.isValidJobType(DEPENDENCY_NAME1, JOB_TYPE_PIG));
+  }
+
+  @Test
+  public void testClone() {
+    executableMap.add(DEPENDENCY_NAME1, DEPENDENCY_VALUE11, JOB_TYPE_SPARK);
+    Assert.assertEquals(1, executableMap.size());
+    ExecutableRampDependencyMap clonedExecutableMap = executableMap.clone();
+    Assert.assertEquals(clonedExecutableMap, executableMap);
```

Comment:
Please swap the params, assertEquals takes (expected, actual). executableMap is expected. clonedExecutableMap is actual.

Reply:
good catch